images/newrelic-nri-statsd/tests/smoke.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+set -o errexit -o nounset -o errtrace -o pipefail -x
+
+container_name="newrelic-nri-statsd-${RANDOM}"
+
+# Step 0: Pull and run a StatsD Docker container
+echo "Starting StatsD Docker container..."
+docker run -d -e NR_API_KEY=${NR_API_KEY}  --name $container_name -p 8125:8125 ${IMAGE_NAME}
```

Comment:
I think we should use $FREE_PORT here to ensure we're always assigning an available port

Reply:
brilliant!